src/lib/aws/common/construct.ts

```diff
@@ -161,4 +162,13 @@ export class CommonConstruct extends Construct {
    * This is determined by the stage property injected via cdk context
    */
   public isProductionStage = () => isPrdStage(this.props.stage)
+
+  protected resolveRef(value: string): string {
+    _.templateSettings.interpolate = /{{([\s\S]+?)}}/g
```

Comment:
This is not maintainable. and we should not be building in Stack output dependencies at common construct level

Reply:
Not sure I fully understand the comment. This does not build any dependencies but rather provides a utility function for other construct to be able to pull data from stack outputs.